Sources/WebAuthn/Ceremonies/Registration/PublicKeyCredentialCreationOptions.swift

```diff
@@ -55,17 +77,87 @@ public struct PublicKeyCredentialCreationOptions: Encodable, Sendable {
         try container.encode(user, forKey: .user)
         try container.encode(relyingParty, forKey: .relyingParty)
         try container.encode(publicKeyCredentialParameters, forKey: .publicKeyCredentialParameters)
-        try container.encodeIfPresent(timeout?.milliseconds, forKey: .timeout)
+        try container.encodeIfPresent(timeoutInMilliseconds, forKey: .timeoutInMilliseconds)
         try container.encode(attestation, forKey: .attestation)
+        try container.encode(authenticatorSelection, forKey: .authenticatorSelection)
+        try container.encode(hints, forKey: .hints)
+        try container.encode(extensions, forKey: .extensions)
+        try container.encode(excludeCredentials, forKey: .excludeCredentials)
+    }
+    
+    let hints: [Hint]
+    enum Hint: String, Encodable
+    {
+        /// Hint to get the user to register their platform authenticator
+        case clientDevice = "client-device"
+        
+        /// Hint to the user should be guided to register a security key. Iconography and text should emphasize the use of security keys
+        case securityKey = "security-key"
+        
+        /// Hint to the user to to register a passkey using their mobile device by scanning a QR code that’s displayed on a computer
+        case hybrid = "hybrid"
+    }
+    
+    let extensions: Extensions
+    
+    struct Extensions: Encodable
+    {
+        let credProps: Bool
+    }
+    
+    let excludeCredentials: [Credentials]
+    struct Credentials: Encodable
+    {
+        let id: String
+        let type: [UInt8]
+        
+        private enum CodingKeys: String, CodingKey
+        {
+            case id
+            case type
+        }
+        
+        public func encode(to encoder: Encoder) throws
+        {
+            var container = encoder.container(keyedBy: CodingKeys.self)
+
+            try container.encode(type.base64URLEncodedString(), forKey: .id)
+            try container.encode(id, forKey: .id)
+        }
+    }
+    
+    let authenticatorSelection: AuthenticatorSelection
+    struct AuthenticatorSelection: Encodable
+    {
+        enum ResidentKey: String, Encodable
+        {
+            case required = "required"
+            case preferred = "preferred"
+            case discouraged = "discouraged"
+        }
+        
+        enum UserVerification: String, Encodable
+        {
+            case required = "required"
+            case preferred = "preferred"
+            case discouraged = "discouraged"
+        }
```

Comment:
Same for these, please use `UnreferencedStringEnumeration`

Reply:
Done as part of `5f8974df64c68188297a6841e77df797bb11ed70`